java.lang.IllegalStateException: Exception thrown on Scheduler.Worker thread. Add `onError` handling.

not sure about the context (AFAIR clicked on My QR code) anymore - but perhaps the stacktrace helps:

```
04-23 22:08:35.172 18550 18550 E AndroidRuntime: java.lang.IllegalStateException: Exception thrown on Scheduler.Worker thread. Add `onError` handling.
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.a.b.b$b.run(LooperScheduler.java:112)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:739)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:95)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:158)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7225)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
04-23 22:08:35.172 18550 18550 E AndroidRuntime: Caused by: rx.exceptions.OnErrorNotImplementedException: Call `Realm.init(Context)` before calling this method.
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.c.e.e$a.a(InternalObservableUtils.java:386)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.c.e.e$a.a(InternalObservableUtils.java:383)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.c.e.b.a(ActionSubscriber.java:44)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.d.a.b(SafeSubscriber.java:153)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.d.a.a(SafeSubscriber.java:115)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.c.a.t$a.a(OperatorObserveOn.java:273)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.c.a.t$a.a(OperatorObserveOn.java:216)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.a.b.b$b.run(LooperScheduler.java:107)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        ... 7 more
04-23 22:08:35.172 18550 18550 E AndroidRuntime: Caused by: java.lang.IllegalStateException: Call `Realm.init(Context)` before calling this method.
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at io.realm.t.n(Realm.java:208)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at com.tokenbrowser.manager.b.b.d(ConversationStore.java:165)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at com.tokenbrowser.manager.v.b(SofaMessageManager.java:251)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at com.tokenbrowser.manager.aj.call(Unknown Source)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.c.a.af.a(SingleFromCallable.java:41)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.c.a.af.a(SingleFromCallable.java:29)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.Single.a(Single.java:1876)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.Single$3$1.a(Single.java:1948)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.c.c.a$b$1.a(CachedThreadScheduler.java:230)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at rx.c.c.i.run(ScheduledAction.java:55)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:423)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:154)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:269)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
04-23 22:08:35.172 18550 18550 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:818)
```

--- a/app/src/main/java/com/tokenbrowser/manager/store/ConversationStore.java
+++ b/app/src/main/java/com/tokenbrowser/manager/store/ConversationStore.java
@@ -170,13 +170,20 @@ public class ConversationStore {
     }
 
     public boolean areUnreadMessages() {
-        final Realm localRealmInstance = Realm.getDefaultInstance();
-        final Conversation result = localRealmInstance
+        final Realm realm;
+        try {
+            realm = Realm.getDefaultInstance();
+        } catch (final IllegalStateException ex) {
+            LogUtil.exception(getClass(), "RealmConfig unexpectedly null", ex);
+            return false;
+        }
+
+        final Conversation result = realm
                 .where(Conversation.class)
                 .greaterThan("numberOfUnread", 0)
                 .findFirst();
         final boolean areUnreadMessages = result != null;
-        localRealmInstance.close();
+        realm.close();
         return areUnreadMessages;
     }
 

--- a/app/src/main/java/com/tokenbrowser/manager/store/PendingMessageStore.java
+++ b/app/src/main/java/com/tokenbrowser/manager/store/PendingMessageStore.java
@@ -21,7 +21,9 @@ package com.tokenbrowser.manager.store;
 import com.tokenbrowser.model.local.PendingMessage;
 import com.tokenbrowser.model.local.SofaMessage;
 import com.tokenbrowser.model.local.User;
+import com.tokenbrowser.util.LogUtil;
 
+import java.util.ArrayList;
 import java.util.List;
 
 import io.realm.Realm;
@@ -43,7 +45,13 @@ public class PendingMessageStore {
 
     // Gets, and removes all messages. After calling this any pending messages will be removed
     public List<PendingMessage> fetchAllPendingMessages() {
-        final Realm realm = Realm.getDefaultInstance();
+        final Realm realm;
+        try {
+            realm = Realm.getDefaultInstance();
+        } catch (final IllegalStateException ex) {
+            LogUtil.exception(getClass(), "RealmConfig unexpectedly null", ex);
+            return new ArrayList<>(0);
+        }
         realm.beginTransaction();
         final RealmResults<PendingMessage> result = realm
                 .where(PendingMessage.class)

